cpu/avr8_common/startup.c

```diff
@@ -57,12 +59,20 @@ void init8_ovr(void) __attribute__((section(".init8")));
 
 __attribute__((used, naked)) void init7_ovr(void)
 {
+#if (ROM_LEN <= 0x2000)
+    __asm__ ("rjmp reset_handler");
+#else
     __asm__ ("call reset_handler");
+#endif
 }
 
 __attribute__((used, naked)) void init8_ovr(void)
 {
+#if (ROM_LEN <= 0x2000)
```

Comment:
This may cause the build issue in the CI because `ROM_LEN` not defined e.g. for atmega256rfr2-xpro (and possibly others) and is evaluated as zero instead.

```suggestion
#if defined(ROM_LEN) && (ROM_LEN <= 0x2000)
```

could be a quick fix. (Or an `#error` when `ROM_LEN` is undefined to ensure every AVR board has this + adding it to the CPUs that don't have it yet.)

Reply:
This will not work even if ROM_LEN is defined in:
```
tests/sys/senml_phydat$ grep -r ROM_LEN ../../../cpu/atmega*
../../../cpu/atmega1281/Makefile.include:ROM_LEN = 128K
../../../cpu/atmega1284p/Makefile.include:ROM_LEN = 128K
../../../cpu/atmega128rfa1/Makefile.include:ROM_LEN = 128K
../../../cpu/atmega2560/Makefile.include:ROM_LEN = 256K
../../../cpu/atmega256rfr2/Makefile.include:ROM_LEN = 256K
../../../cpu/atmega328p/Makefile.include:ROM_LEN = 32K
tests/sys/senml_phydat$ BOARD=atmega1284p make -j64
Building application "tests_senml_phydat" for "atmega1284p" with MCU "atmega1284p".

"make" -C /home/hugues/github/cpu_avr8_common/RIOT/boards/atmega1284p
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/boards/common/init
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/core
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/core/lib
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/cpu/atmega1284p
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/drivers
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/boards/common/atmega
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/cpu/atmega_common
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/drivers/periph_common
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/auto_init
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/embunit
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/fmt
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/isrpipe
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/libc
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/malloc_thread_safe
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/phydat
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/preprocessor
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/cpu/atmega_common/periph
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/senml
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/cpu/avr8_common
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/stdio_uart
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/test_utils/interactive_sync
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/test_utils/print_stack_usage
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/tiny_strerror
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/sys/tsrb
"make" -C /home/hugues/github/cpu_avr8_common/RIOT/cpu/avr8_common/avr_libc_extra
/home/hugues/github/cpu_avr8_common/RIOT/tests/sys/senml_phydat/bin/atmega1284p/avr8_common/startup.o: In function `init8_ovr':
/home/hugues/github/cpu_avr8_common/RIOT/cpu/avr8_common/startup.c:72:(.init8+0x0): relocation truncated to fit: R_AVR_13_PCREL against symbol `exit' defined in .fini9 section in /usr/lib/gcc/avr/5.4.0/avr51/libgcc.a(_exit.o)
collect2: error: ld returned 1 exit status
make: *** [/home/hugues/github/cpu_avr8_common/RIOT/tests/sys/senml_phydat/../../../Makefile.include:750 : /home/hugues/github/cpu_avr8_common/RIOT/tests/sys/senml_phydat/bin/atmega1284p/tests_senml_phydat.elf] Erreur 1
```
I replaced ROM_LEN by FLASHEND